OMCompiler/Compiler/SimCode/SimCodeUtil.mo

```diff
@@ -5584,6 +5587,38 @@ algorithm
   end match;
 end extractIdAndExpFromDelayExp;
 
+function extractSpatialDistributionInfo
+  "Create spatialDistribution simCode from backend DAE."
+  input BackendDAE.BackendDAE dlow;
+  output SimCode.SpatialDistributionInfo spatialInfo;
+protected
+  list<SimCode.SpatialDistribution> spatial_lst;
+  Mutable<Integer> maxIndex_ptr = Mutable.create(-1);
+algorithm
+  ((_,spatial_lst)) := BackendDAEUtil.traverseBackendDAEExps(dlow, Expression.traverseSubexpressionsHelper, (function extractSpatialDistributionInfoExp(maxIndex_ptr = maxIndex_ptr), {}));
+  spatialInfo := SimCode.SPATIAL_DISTRIBUTION_INFO(spatial_lst, Mutable.access(maxIndex_ptr));
+end extractSpatialDistributionInfo;
+
+function extractSpatialDistributionInfoExp
+  input output DAE.Exp callExp;
+  input output list<SimCode.SpatialDistribution> spatialInfo;
+  input Mutable<Integer> maxIndex_ptr;
+algorithm
+  spatialInfo := match callExp
+    local
+      Integer i, initSize;
+      DAE.Exp in0, in1, pos, dir, initPnts, initVals;
+    case DAE.CALL(path = Absyn.IDENT("spatialDistribution"), expLst={DAE.ICONST(i), in0, in1, pos, dir, initPnts, initVals})
+      algorithm
+        if i > Mutable.access(maxIndex_ptr) then
+          Mutable.update(maxIndex_ptr, i);
+        end if;
+        initSize := Expression.sizeOf(Expression.typeof(initPnts));
```

Comment:
Is it safe to assume at this point that `initPnts` and `initVals` have the same size?
If not, we can test it here maybe.
Otherwise why do we need `initSize`, can't we compute it in codegen directly?

Reply:
I'm not sure where the best place is to test that. @kabdelhak any idea?
But we should check this way more early then here. In theory the Frontend should be able to verify at checkModel if initialPoints and initialValues have the same size, are in [0,1], don't have to  many events on one point and so on.

Reply:
I added an error here for now. Double checking is not a bad idea.

Reply:
The check is not expensive so sure why not. I think that is mainly frontend work, but maybe it gets manipulated with replacements and we want to detect errors.